lib/file-store.js

```diff
@@ -43,7 +42,40 @@ module.exports.upload = async (display_filename, contents, type, assessment_inst
         instance_question_id,
         user_id,
         authn_user_id,
+        storage_type,
     ];
+
+    // uploads to ec2 filesystem
+    if (storage_type === this.storageTypes.FileSystem) {
+        debug('upload()');
+        // Make a filename to store the file. We use a UUIDv4 as the filename,
+        // and put it in two levels of directories corresponding to the first-3
+        // and second-3 characters of the filename.
+
+        if (config.filesRoot == null) throw new Error('config.filesRoot is null, which does not allow uploads');
+
+        debug(`upload() : mkdir ${dir}`);
+        await fsPromises.mkdir(dir, {recursive: true, mode: 0o700});
+        debug(`upload(): writeFile ${filename}`);
+        await fsPromises.writeFile(filename, buffer, {mode: 0o600});
+    }
```

Comment:
Is this for the "attach a file" dialog?  I'm actually thinking we can remove local EC2 file storage as that will just cause problems when we're running on multiple servers.  On local dev, we can probably just use s3rver as well. (Thoughts, @mwest1066 ?)

Reply:
@nicknytko "Attach a file" dialog is right. Keep in mind that the instructor file editor also goes through this path, so the files get uploaded to S3. That may be something you would prefer the FileSystem for. In my tests, S3 was quite fast though.

For something like the instructor file editor, are we happy with the history of the files being left on S3? I cannot help but notice the soft deletes.

Whichever approach you want to take should work. They both work in the manual and CI tests.

Reply:
The get() function will need to continue retrieving FileSystem files that already exist.

Reply:
I think sending everything to S3 is fine. Keeping a history in S3 is ok (we can always clean these up later if necessary).

@nicknytko I agree that we want to switch everything to S3, but we have a large set of files that need to be migrated from the existing disk store. My thinking was to get this PR merged and default to putting new files in S3, then at some point we can migrate the existing files off the disk.